File: TransCoder_tokenized_test_set_functions/java/PRINTING_STRING_PLUS_PATTERN_MATRIX.java
static void carveCross ( String str ) {
  int n = str . length ( );
  if ( n % 2 == 0 ) {
    System . out . print ( "Not possible. Please enter " + "odd length string.\n" );
  }
  else {
    char arr [ ] [ ] = new char [ max ] [ max ];
    int m = n / 2;
    for ( int i = 0;
    i < n;
    i ++ ) {
      for ( int j = 0;
      j < n;
      j ++ ) {
        arr [ i ] [ j ] = 'X';
      }
    }
    for ( int i = 0;
    i < n;
    i ++ ) {
      arr [ i ] [ m ] = str . charAt ( i );
    }
    for ( int i = 0;
    i < n;
    i ++ ) {
      arr [ m ] [ i ] = str . charAt ( i );
    }
    for ( int i = 0;
    i < n;
    i ++ ) {
      for ( int j = 0;
      j < n;
      j ++ ) {
        System . out . print ( arr [ i ] [ j ] + " " );
      }
      System . out . print ( "\n" );
    }
  }
}

File: TransCoder_tokenized_test_set_functions/python/PRINTING_STRING_PLUS_PATTERN_MATRIX.py
def carveCross ( str ) :
    n = len ( str )
    if ( n % 2 == 0 ) :
        print ( "Not possible. Please enter ", "odd length string.\n" )
    else :
        arr = [ [ False for x in range ( max ) ] for y in range ( max ) ]
        m = n //2
        foriinrange(n):
            forjinrange(n):
                arr[i][j]='X'
        foriinrange(n):
            arr[i][m]=str[i]
        foriinrange(n):
            arr[m][i]=str[i]
        foriinrange(n):
            forjinrange(n):
                print(arr[i][j],end=" ")
            print()
    
